rethinkdb/check.py

```diff
@@ -0,0 +1,307 @@
+    "queries_total": ('rethinkdb.server.queries_total', GAUGE),
+    "read_docs_per_sec": ('rethinkdb.server.read_docs_per_sec', GAUGE),
+    "read_docs_total": ('rethinkdb.server.read_docs_total', GAUGE),
+    "written_docs_per_sec": ('rethinkdb.server.written_docs_per_sec', GAUGE),
+    "written_docs_total": ('rethinkdb.server.written_docs_total', GAUGE),
+    "client_connections": ('rethinkdb.server.client_connections', GAUGE),
+    "clients_active": ('rethinkdb.server.clients_active', GAUGE)}
+
+REPLICATION_METRICS = {
+    "read_docs_per_sec": ('rethinkdb.repl.read_docs_per_sec', GAUGE),
+    "read_docs_total": ('rethinkdb.repl.read_docs_total', GAUGE),
+    "written_docs_per_sec": ('rethinkdb.repl.written_docs_per_sec', GAUGE),
+    "written_docs_total": ('rethinkdb.repl.written_docs_total', GAUGE),
+    "in_use_bytes" : ('rethinkdb.repl.cache.in_use_bytes', GAUGE),
+    "read_bytes_per_sec": ('rethinkdb.repl.disk.read_bytes_per_sec', GAUGE),
+    "read_bytes_total": ('rethinkdb.repl.disk.read_bytes_total', GAUGE),
+    "written_bytes_per_sec": ('rethinkdb.repl.disk.written_bytes_per_sec', GAUGE),
+    "written_bytes_total": ('rethinkdb.repl.disk.written_bytes_total', GAUGE),
+    "metadata_bytes": ('rethinkdb.repl.space_usage.metadata_bytes', GAUGE),
+    "data_bytes": ('rethinkdb.repl.space_usage.data_bytes', GAUGE),
+    "garbage_bytes": ('rethinkdb.repl.space_usage.garbage_bytes', GAUGE),
+    "preallocated_bytes" : ('rethinkdb.repl.space_usage.preallocated_bytes', GAUGE)}
+
+TABLE_CONFIG_METRICS = {
+    "shards_count": ('rethinkdb.table.config.shards_count', GAUGE),
+    "replicas_count": ('rethinkdb.table.config.replicas_count', GAUGE),
+    "index_count": ('rethinkdb.table.config.index_count', GAUGE),
+    "durability": ('rethinkdb.table.config.durability.code', GAUGE)}
+
+DURABILITY_CODE = {
+    "soft": 0,
+    "hard": 1,
+    "other": 3}
+
+class RethinkdbCheck(AgentCheck):
+    hostname = ''
+
+    def check(self, instance):
+        self.log.info('inside rethinkdb check')
+        hostname = instance.get('host','localhost')
+        self.hostname = hostname.strip()
+        port = instance.get('port')
+        collect_extra = instance.get('additional_metrics',True)
+        if not all([hostname,port]):
+            raise Exception('The Rethinkdb hostname and password must be specified in the configuration')
+
+        instance_tags = ["rethinkdb_instance:{0}-{1}".format(hostname, port)]
+
+        #service check
+        self.service_check(RETHINKDB_SERVICE_CHECK,
+            AgentCheck.OK,
+            tags=instance_tags,
+            message='Connection to %s was successful' % hostname)
+
+        # get and set metrics
+        self._get_rethinkdb_metrics(hostname, port, instance_tags, collect_extra)
+
+    def _get_connection(self, hostname, port, instance_tags):
+        '''
+        Get connection to rethinkdb
+        '''
+        try:
+            conn = r.connect(host = hostname, port = port)
```

Comment:
If user/pass are specified, try to connect with that first, and if it fails, try to connect without it in case it's an older version that does not support auth

```
the user auth system was introduced in version 2.3, so if the rethinkdb.user table does not exist then the user/password option can be ignored ```